pkg/util/common.go

```diff
@@ -19,6 +22,64 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/version"
 )
 
+// CopyFile atomically copies file path `src`` to file path `dst`.
+func CopyFile(src, dst string) error {
+	fi, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+	perm := fi.Mode()
+
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+
+	tmp, err := ioutil.TempFile(filepath.Dir(dst), "")
+	if err != nil {
+		return err
+	}
+	tmpName := tmp.Name()
+
+	_, err = io.Copy(tmp, in)
+	if err != nil {
+		tmp.Close()
+		os.Remove(tmpName)
+		return err
+	}
+
+	err = tmp.Close()
+	if err != nil {
+		os.Remove(tmpName)
+		return err
+	}
+
+	err = os.Chmod(tmpName, perm)
+	if err != nil {
+		os.Remove(tmpName)
+		return err
+	}
+
+	err = os.Rename(tmpName, dst)
+	if err != nil {
+		os.Remove(tmpName)
+		return err
+	}
+
+	return nil
+}
+
+// CopyFileAll calls CopyFile, but will create necessary directories for  `dst`.
+func CopyFileAll(src, dst string) error {
+	err := os.MkdirAll(filepath.Dir(dst), 0644)
```

Comment:
Same here, `0644` won't behave as you expect on Windows, you should consider moving these functions behind a build flag.